baybe/simulation/lookup.py

```diff
@@ -83,6 +83,8 @@ def _look_up_targets_from_callable(
     """Look up target values by querying a callable."""
     df_targets = lookup(queries)
     queries[df_targets.columns] = df_targets.values
+    # Esnure that the data types in queries are the same as the ones in df_targets
+    queries[df_targets.columns] = queries[df_targets.columns].astype(df_targets.dtypes)
```

Comment:
The actual source of the problem is because going via the numpy layer loses the column metadata information, i.e. the entire array is then represented using the supertype common to all columns, which destroys the float representation whenever there is a single `object` contained. Instead of applying two unnecessary conversions back and forth, the following should do
```suggestion
    queries[df_targets.columns] = df_targets
```